yamux/src/connection/stream.rs

```diff
@@ -367,18 +369,19 @@ impl AsyncWrite for Stream {
             let k = std::cmp::min(shared.credit as usize, buf.len());
             let k = std::cmp::min(k, self.config.split_send_size);
             shared.credit = shared.credit.saturating_sub(k as u32);
-            Vec::from(&buf[..k])
+            &buf[..k]
         };
         let n = body.len();
         let mut frame = Frame::data(self.id, body).expect("body <= u32::MAX").left();
-        self.add_flag(frame.header_mut());
+        let mut parsed_frame = frame.parse_mut().expect("valid frame");
+        self.add_flag(parsed_frame.header_mut());
         log::trace!("{}/{}: write {} bytes", self.conn, self.id, n);
 
         // technically, the frame hasn't been sent yet on the wire but from the perspective of this data structure, we've queued the frame for sending
         // We are tracking this information:
         // a) to be consistent with outbound streams
         // b) to correctly test our behaviour around timing of when ACKs are sent. See `ack_timing.rs` test.
-        if frame.header().flags().contains(ACK) {
+        if parsed_frame.header().flags().contains(ACK) {
```

Comment:
Parsing the frame is cheap I'd assume? Can we make this more ergonomic by adding a `has_flag` function to `Frame` so we don't need to keep around the `parsed_frame` here?

Reply:
add_flag() now returns the flags, so no need to call contains again